src/classes/INT_InteractionProcessor.cls

```diff
@@ -307,57 +302,52 @@ public class INT_InteractionProcessor {
-                    leadsToConvert.add(createLeadConvert(interactionMap.get(intearctionId), leadId, true));
-                    interaction.Lead__c = leadId;
-                    leadIdToInteractionMap.put(interaction.Lead__c, interaction);
+            if (sr.isSuccess()) { // Check if insert was a success
+                // On success, prepare to convert inserted lead.
+                Id leadId = sr.getId();
+                leadsToConvert.add(createLeadConvert(interactionMap.get(interaction.Id), leadId, true));
+                interaction.Lead__c = leadId; // Copy new Lead Id to the Interaction__c record.
+                if (!interaction.Lead_Only__c) leadsToDelete.add(new Lead(Id = leadId)); // Newly created leads will be deleted
+            } else {
+                /**
+                 * If there are errors, loop through them and either log the error on the Interaction.
+                 * or see if there's a duplicate and use that instead.
+                 */
+                for (Database.Error error : sr.getErrors()) {
+                    if (error.getStatusCode() == StatusCode.DUPLICATES_DETECTED) {
+                        Database.DuplicateError dupErrorError = (Database.DuplicateError) error;
 
-                    if (interaction.Lead_Only__c == false) {
-                        leadsToDelete.add(new Lead(Id = leadId)); // Newly created leads will be deleted
-                    }
-                } else {
-                    /**
-                     * If there are errors, loop through them and either log the error on the Interaction.
-                     * or see if there's a duplicate and use that instead.
-                     */
-                    for (Database.Error error : sr.getErrors()) {
-                        if (error.getStatusCode() == StatusCode.DUPLICATES_DETECTED) {
-                            Database.DuplicateError dupErrorError = (Database.DuplicateError) error;
-
-                            for (Datacloud.MatchResult matchResult : dupErrorError.getDuplicateResult().getMatchResults()) {
-                                for (Datacloud.MatchRecord match : matchResult.getMatchRecords()) {
-                                    // Grab matched record to use instead of the new one.
-                                    leadIds.add(match.getRecord().Id);
-                                    interaction.Lead__c = match.getRecord().Id;
-                                    leadIdToInteractionMap.put(interaction.Lead__c, interaction);
-
-                                    if (interaction.Lead_Only__c == false) {
+                        for (Datacloud.MatchResult matchResult : dupErrorError.getDuplicateResult().getMatchResults()) {
+                            for (Datacloud.MatchRecord match : matchResult.getMatchRecords()) {
+                                // Grab matched record to use instead of the new one.
+                                interaction.Lead__c = match.getRecord().Id;
```

Comment:
@karamcintyre here is where I made the single line change to copy the Lead Id to the Interaction if a Lead existed in the system already.